leases/utils.py

```diff
@@ -63,3 +63,30 @@ def calculate_berth_lease_end_date() -> date:
     Leases always end on 14.9, the year depends on the current year
     """
     return calculate_season_end_date()
+
+
+def calculate_winter_storage_lease_start_date() -> date:
+    """
+    Return the date when the "winter" lease season should start
+
+    The season should start by default on 15.9 (one day after the "summer" season ends)
+    """
+
+    # Otherwise, return the latest date between the default start date or today
+    today = date.today()
+    default = max(date(day=15, month=9, year=today.year), today)
+    return default
+
+
+def calculate_winter_storage_lease_end_date() -> date:
+    """Return the date when the lease season should end
+
+    The season should end by default on 10.6 on the following year
+    """
+    today = date.today()
+    default = date(day=10, month=6, year=today.year + 1)
+
+    if today.month < default.month:
```

Comment:
and here we probably should also compare the days? otherwise leases generated from 1st till 10th of June will go to the next year 🤔

Reply:
🤔 Tru, good catch

Reply:
extreme edge case, but since we noticed, we can account for it already now 😅

Reply:
Yup yup, better safe than sorry 😛